### PR TITLE
refactor!: cleanup deprecations

### LINKED ISSUE #1439: Remove `@deprecated` code
Remove code marked with `@deprecated`

### PR DESCRIPTION
**Pull request by Shinigami92**
closes #1439

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -55,13 +54,12 @@ export type { InternetModule } from './modules/internet';
 export type { LoremModule } from './modules/lorem';
 export type { MersenneModule } from './modules/mersenne';
```
Comment (ST-DDT) [inline src/index.ts]: This shouldn't be exported anymore as it is internal.

File: test/address.spec.ts [hunk 2]
```diff
@@ -54,12 +54,9 @@ describe('address', () => {
...
     });
 
-    t.it('cityName')
-      .it('cityPrefix')
-      .it('citySuffix')
-      .describe('city', (t) => {
-        t.it('noArgs').it('with given index', 1);
-      });
+    t.it('cityName').describe('city', (t) => {
+      t.it('noArgs').it('with given index', 1);
```
Comment (ST-DDT) [inline test/address.spec.ts]: `city(index)` is deprecated and should be removed (also from src).

File: test/company.spec.ts [hunk 3]
```diff
@@ -22,10 +22,7 @@ describe('company', () => {
       'bsBuzz',
       'bsNoun'
     );
-    t.describeEach(
-      'companyName',
-      'name'
-    )((t) => {
+    t.describeEach('name')((t) => {
       t.it('noArgs').it('with index');
```
Comment (ST-DDT) [inline test/company.spec.ts]: `company(index)` is deprecated and should be removed also from `src`.

File: test/phone.spec.ts [hunk 4]
```diff
@@ -11,33 +11,15 @@ describe('phone', () => {
   });
 
   seededTests(faker, 'phone', (t) => {
-    t.itEach('phoneFormats', 'imei');
+    t.itEach('imei');
```
Comment (ST-DDT) [inline test/phone.spec.ts]: Maybe just `t.it()`?

File: test/phone.spec.ts [hunk 5]
```diff
@@ -11,33 +11,15 @@ describe('phone', () => {
...
 
-    t.describeEach(
-      'phoneNumber',
-      'number'
-    )((t) => {
+    t.describeEach('number')((t) => {
```
Comment (ST-DDT) [inline test/phone.spec.ts]: Maybe just `t.describe()`?

**Review by ST-DDT — CHANGES_REQUESTED** [review]

File: src/modules/address/index.ts [hunk 6]
```diff
@@ -69,82 +69,19 @@ export class AddressModule {
...
-        until: '8.0',
-      });
-    }
+  city(): string {
     const formats = this.faker.definitions.address.city;
 
-    if (typeof format !== 'number') {
-      format = this.faker.datatype.number(formats.length - 1);
-    }
+    const format = this.faker.datatype.number(formats.length - 1);
```
Comment (ST-DDT) [inline src/modules/address/index.ts]: Please use `helpers.arrayElement` instead.

File: src/modules/company/index.ts [hunk 7]
```diff
@@ -57,38 +45,11 @@ export class CompanyModule {
     const formats =
       this.faker.definitions.company.name_patterns ?? staticFormats;
 
-    if (typeof format !== 'number') {
-      format = this.faker.datatype.number(formats.length - 1);
-    }
+    const format = this.faker.datatype.number(formats.length - 1);
```
Comment (ST-DDT) [inline src/modules/company/index.ts]: Please use `helpers.arrayElement` instead.

**Review by ST-DDT — CHANGES_REQUESTED** [review]

File: src/modules/address/index.ts (hunk 6 above)
Comment (Shinigami92) [reply src/modules/address/index.ts]: nice :slightly_smiling_face:

File: src/modules/company/index.ts (hunk 7 above)
Comment (Shinigami92) [reply src/modules/company/index.ts]: nice :slightly_smiling_face:

**Review by ST-DDT — DISMISSED** [review]

**Comment by xDivisionByZerox** [comment]
How do we handle this as it is marked for removal in v8.0?
https://github.com/faker-js/faker/blob/2914a69f7da9552fd0a58c57d2961e4136dfa7b2/src/modules/address/index.ts#L137-L152

**Comment by xDivisionByZerox** [comment]
You missed this on:
https://github.com/faker-js/faker/blob/2914a69f7da9552fd0a58c57d2961e4136dfa7b2/src/modules/datatype/index.ts#L220-L235

**Review by xDivisionByZerox — CHANGES_REQUESTED** [review]
You missed this on:
https://github.com/faker-js/faker/blob/2914a69f7da9552fd0a58c57d2961e4136dfa7b2/src/modules/datatype/index.ts#L220-L235

**Review by ejcheng — DISMISSED** [review]

**Comment by ejcheng** [comment]
> You missed this on:
> 
> https://github.com/faker-js/faker/blob/2914a69f7da9552fd0a58c57d2961e4136dfa7b2/src/modules/datatype/index.ts#L220-L235

Good catch.

**Comment by Shinigami92** [comment]
@xDivisionByZerox nice catch!
I also found streetName :+1:

**Review by xDivisionByZerox — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Pull request merged**